blocks/product-support/product-support.js

```diff
@@ -0,0 +1,152 @@
+import { createDomStructure, translate } from '../../scripts/lib-franklin.js';
+
+function getInfo() {
+  const url = new URL(window.location);
+  const idx = url.pathname.indexOf('/product-support/');
+  if (idx > 0) {
+    const slug = url.pathname.substring(url.pathname.indexOf('/product-support/') + '/product-support/'.length);
+    if (slug) {
+      return {
+        productCode: slug,
+        productSupport: url.pathname.substring(0, url.pathname.indexOf(`${slug}`)),
+        language: url.pathname.substring(1, url.pathname.indexOf('/', 1)),
+      };
+    }
+  }
+  return { productSupport: url.pathname.substring(0, url.pathname.length - 1) };
+}
+
+function getProduct(json, productCode, language) {
```

Comment:
could probably be made more concise like this:
```
function getProduct(json, productCode, language) {
  const product = json.Product.data.find((entry) => {
    const codeExists = entry.ProductCodes.split('|').includes(productCode);
    const langExists = entry.Languages.split('|').map((lang) => lang.toUpperCase()).includes(language.toUpperCase());
    return codeExists && langExists;
  });

  if (product) {
    const translation = json.ProductTranslation.data.find((entry) => entry.ProductRef === product.ProductCodes && entry.Language === language);
    if (translation) {
      product.Name = translation.Name || product.Name;
      product.Image = translation.Image || product.Image;
    }
  }

  return product;
}
```

Reply:
Done. I removed the if block as well by using  a question mark